py/server/deephaven/table.py

```diff
@@ -1285,3 +1274,132 @@ def layout_hints(self, front: Union[str, List[str]] = None, back: Union[str, Lis
+        """ Creates a PartitionedTable from this table, partitioned according to the specified key columns.
+
+        Args:
+            table (Table): the source table
+            by (Union[str, Sequence[str]]): the column(s) by which to group data
+            drop_keys (bool): whether to drop key columns in the constituent tables, default is False
+
+        Returns:
+            A PartitionedTable containing a sub-table for each group
+
+        Raises:
+            DHError
+        """
+        try:
+            by = to_sequence(by)
+            return PartitionedTable(j_partitioned_table=self.j_table.partitionBy(drop_keys, *by))
+        except Exception as e:
+            raise DHError(e, "failed to create a partitioned table.") from e
+
+
+class PartitionedTable(JObjectWrapper):
+    """A partitioned table is a table with one column containing like-defined constituent tables,
+    optionally with key columns defined to allow binary operation based transformation or joins with other like-keyed
+    partitioned tables."""
+
+    j_object_type = _JPartitionedTable
+
+    @property
+    def j_object(self) -> jpy.JType:
+        self.j_partitioned_table
+
+    def __init__(self, j_partitioned_table):
+        self.j_partitioned_table = j_partitioned_table
+        self._schema = None
+
+    @property
+    def table(self) -> Table:
+        """The underlying Table."""
+        return Table(j_table=self.j_partitioned_table.table())
+
+    @property
+    def key_columns(self) -> List[str]:
+        """The partition key column names."""
+        return list(self.j_partitioned_table.keyColumnNames().toArray())
+
+    @property
+    def unique_keys(self) -> bool:
+        """Whether the keys in the underlying table are unique."""
+        return self.j_partitioned_table.uniqueKeys()
+
+    @property
+    def constituent_column(self) -> str:
+        """The constituent column name."""
+        return self.j_partitioned_table.constituentColumnName()
+
+    @property
+    def constituent_table_columns(self) -> List[Column]:
+        """The column definitions shared by the constituent tables."""
+        if not self._schema:
+            self._schema = _td_to_columns(self.j_partitioned_table.constituentDefinition())
+
+        return self._schema
+
+    @property
+    def constituent_change_permitted(self) -> bool:
+        """Whether the constituents of the underlying partitioned table can change."""
+        return self.j_partitioned_table.constituentChangesPermitted()
+
+    def merge(self) -> Table:
+        """Makes a new Table that contains all the rows from all the constituent tables."""
+        try:
+            return Table(j_table=self.j_partitioned_table.merge())
+        except Exception as e:
+            raise DHError(e, "failed to merge all the constituent tables.")
+
+    def filter(self, filters: Union[Filter, Sequence[Filter]]) -> PartitionedTable:
```

Comment:
Are we still pending wrappers for `Filter`?

Reply:
Filter is wrapped.

Reply:
I guess I was trying to explain to myself why filter and sort are validated so differently.